hadoop-hdfs-project/hadoop-hdfs/src/main/webapps/hdfs/explorer.js

```diff
@@ -518,7 +518,8 @@
             url: url,
             data: file.file,
             processData: false,
-            crossDomain: true
+            crossDomain: true,
+            contentType: 'application/octet-stream'
```

Comment:
Question: Is this a js running in client browsers?  The change here is to change the `contentType` when uploading a file from a client browser to HDFS.  Is it correct?

Reply:
@szetszwo Yes, this is the JS code running in the client browser. By default the contentType is set to `application/x-www-form-urlencoded` now it's explicitly set to `application/octet-stream`.

However I a found a problem when testing with newer browsers with CORS enabled. It works when I try it via Knox or with a browser where CORS is disabled.

But in new browsers with Cross-Origin Resource Sharing policy prevents changing the contentType since it's not included in the `Access-Control-Allow-Headers` header. Only the accept header is included in `Access-Control-Allow-Headers`.

```
Access-Control-Allow-Headers: Accept
Access-Control-Allow-Methods: PUT
Access-Control-Allow-Origin: *
``` 

The CrossOriginFilter contains the content type so I don't know why I only see `Accept`

 https://github.com/apache/hadoop/blob/9de13f879af66a211b098166fd423b6900fc730d/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/http/CrossOriginFilter.java#L74

Reply:
The list is the default values.  It can be overridden by init parameter; see
https://github.com/apache/hadoop/blob/9de13f879af66a211b098166fd423b6900fc730d/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/http/CrossOriginFilter.java#L176-L185

Reply:
Yes I've seen that, but it wasn't overridden. Even if I explicitly added   

```xml
<property>
    <name>hadoop.http.cross-origin.allowed-headers</name>
    <value>X-Requested-With,Content-Type,Accept,Origin</value>
  </property>
```

it was not picked up for some reason.

Reply:
@szetszwo I'm not sure the CORS problem still exists. I've tested this using Chrome browser and with maximum security enabled the change was still working.